plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureHttpHandler.java

```diff
@@ -0,0 +1,250 @@
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+package org.elasticsearch.repositories.azure;
+
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.regex.Regex;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.RestUtils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Minimal HTTP handler that acts as an Azure compliant server
+ */
+public class AzureHttpHandler implements HttpHandler {
+
+    private final Map<String, BytesReference> blobs;
+    private final String container;
+
+    public AzureHttpHandler(final String container) {
+        this.container = Objects.requireNonNull(container);
+        this.blobs = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public void handle(final HttpExchange exchange) throws IOException {
+        final String request = exchange.getRequestMethod() + " " + exchange.getRequestURI().toString();
+
+        if (request.startsWith("GET") || request.startsWith("HEAD") || request.startsWith("DELETE")) {
+            int read = exchange.getRequestBody().read();
+            assert read == -1 : "Request body should have been empty but saw [" + read + "]";
+        }
+        try {
+            if (Regex.simpleMatch("PUT /" + container + "/*blockid=*", request)) {
+                // Put Block (https://docs.microsoft.com/en-us/rest/api/storageservices/put-block)
+                final Map<String, String> params = new HashMap<>();
+                RestUtils.decodeQueryString(exchange.getRequestURI().getQuery(), 0, params);
```

Comment:
Could you update the branch against master? I recently deleted this method as it was unused.

Reply:
Shall i add it back ?

Reply:
Just wanted to make sure I see the full diff. But now I'm actually wondering if we could replace it with `io.netty.handler.codec.http.QueryStringDecoder` or so?